app.ts

```diff
@@ -866,7 +870,8 @@ async function main() {
             );
         })
         .use(bodyParser.json({limit: ceProps('bodyParserLimit', maxUploadSize)}))
-        .use('/source', sourceHandler.handle.bind(sourceHandler))
+        .use('/source/:source/list', sourceHandler.listEntries.bind(sourceHandler))
+        .use('/source/:source/load/:language/:filename', sourceHandler.loadEntry.bind(sourceHandler))
```

Comment:
Full paths (meaning we get params from express) now, instead of a catch-all route

Reply:
Much preferred, thanks. You're really unearthing my earliest code here! :) [ https://github.com/compiler-explorer/compiler-explorer/blob/15ea5e164b55c2b5ee0d3b432e3984b8f361afd2/app.js#L169-L180 literally in the initial checkin)

Reply:
oh my that's 12 years ago